### ISSUE TITLE
OOM error at WiProProtocol.MessageFrameAssembler.handleFirstDataFrame()

### ISSUE THREAD

**Issue by askirk** [smartdevicelink/sdl_android#389]
Stacktrace:
```
Fatal Exception: java.lang.OutOfMemoryError
       at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1068)
       at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1063)
Caused by java.lang.OutOfMemoryError: Failed to allocate a 587661569 byte allocation with 4070898 free bytes and 244MB until OOM
       at java.io.ByteArrayOutputStream.(ByteArrayOutputStream.java)
       at com.smartdevicelink.protocol.WiProProtocol$MessageFrameAssembler.handleFirstDataFrame(SourceFile:11264)
       at com.smartdevicelink.protocol.WiProProtocol$MessageFrameAssembler.handleFrame(SourceFile:348)
       at com.smartdevicelink.protocol.WiProProtocol.handlePacketReceived(SourceFile:226)
```

Occurs at https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android_lib/src/com/smartdevicelink/protocol/WiProProtocol.java#L264

Tries to allocate 587 MB in first data frame.

**Comment by askirk** [comment]
Try/catch at L264 
`if(accumulator!=null)' @ https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android_lib/src/com/smartdevicelink/protocol/WiProProtocol.java#L312

**Comment by joeygrover** [comment]
Fixed in #407
